### ISSUE TITLE
Doesn't work on Raspberry Pi 2

### ISSUE THREAD

**Issue by Gadgetoid** [626Pilot/RaspberryPi-NeoPixel-WS2812#4]
Ahoy! Just to let you now that this code now doesn't work on the Raspberry Pi 2. We use it in the library to drive one of our products, and I've been struggling to understand and fix it. I haven't got anywhere, so this is a humble cry for help! We'd be happy to support your efforts to make it Pi 2 compatible if you're up for it!

**Comment by laboratories** [comment]
yes please sir, more Raspberry Pi 2, please

**Comment by arthur-esob** [comment]
i was about to purchase a neopixel until I ran across your post Gadgetoid :-( would it help if i buy one and be a "guinea pig" ? I'm really looking forward to using the neopixel ring so if i can help...

**Comment by Corteil** [comment]
They work on Raspberry Pi 2 model b, have a look on my blog for details. http://www.corteil.co.uk/Blog/sonic-pi-trophies/

Have  fun

Brian C

**Comment by arthur-esob** [comment]
ok thanks for the information. i just finished ordering a neopixel. ar you in for any kind of updates ?

**Comment by arthur-esob** [comment]
arfff... just received my neopixel, followed your guide on the blog and i get an error ... here's a snapshot that will help i hope ...
pierre@raspberrypi ~/neopixel/rpi_ws281x/python $ ls
examples  ez_setup.py  ez_setup.pyc  neopixel.py  neopixel.pyc  rpi_ws281x.i  setup.py  strandtest.py
pierre@raspberrypi ~/neopixel/rpi_ws281x/python $ cd examples/
pierre@raspberrypi ~/neopixel/rpi_ws281x/python/examples $ ls
lowlevel.py  neopixel.py  neopixel.pyc  strandtest.py
pierre@raspberrypi ~/neopixel/rpi_ws281x/python/examples $ sudo python strandtest.py 
Traceback (most recent call last):
  File "strandtest.py", line 9, in <module>
    from neopixel import *
  File "/home/pierre/neopixel/rpi_ws281x/python/examples/neopixel.py", line 3, in <module>
    import _rpi_ws281x as ws
ImportError: No module named _rpi_ws281x

doesn't seem to find the    _rpi_ws281x   module and i really don't know why ... any idea ??
pierre

**Comment by arthur-esob** [comment]
the 'test' program in rpi_ws281x lights almost all diodes (9 out of 12) ... i keep searching

**Comment by Corteil** [comment]
Glad you found the answer.

**Comment by communitywireless** [comment]
This does not work on Raspberry Pi 2 B v 1.1.  When I run the strandtest.py the session hangs and the python process needs to be killed from another terminal.

The EXACT same CF, LED strips and breadboard layout on a Raspberry Pi B+ 512MB runs fine.

**Comment by Gadgetoid** [comment]
@communitywireless I raised this in February, and nobody has yet got this particular library to work on Pi 2- *however* there is now an alternative which uses a different method and works beautifully: https://github.com/richardghirst/rpi_ws281x

Since this library hasn't been fixed, and from what I understand cannot be fixed ( we spent a few weeks trying ), then this issue is redundant. See the link above.

**Comment by mclien** [comment]
> import _rpi_ws281x as ws
> ImportError: No module named _rpi_ws281x

Mind to share how you fixed that? I ran in the same problem and have no idea how to fix it...

**Comment by arthur-esob** [comment]
Wow it's been a while. 
Knowing myself, I must have tried to find the piece of code that was problematic and try to figure out how to fixe this.
Let me know if you have any luck.  I ll try to remember how i did that....

Date: Mon, 15 Aug 2016 09:16:54 -0700

**Comment by mclien** [comment]
Because I ran into deadline problems, so I needed a quick fix, a friend helped out and we solved it with this:
https://www.pjrc.com/teensy/td_libs_OctoWS2811.html
The Raspi is delivering the data via USB.

Additionally I was lagging a IC to get the 3.3V signal from the raspi stepped up to 5V.
(We had some lights, but nothing correct). So perhaps I could have worked with the right voltage step up. I'll try to figure that out, but it's all kinda blurry right now after 2 nights with about 3h sleep max...

**Comment by Graystripe17** [comment]
The suggested library appears to be behind the original. Does anyone know what it is missing?

**Comment by Graystripe17** [comment]
When performing the exact same steps to run test it says "Can't open device file /tmp/mailbox-3525: No such device or address"
Is this an issue that I should open up?

**Comment by Graystripe17** [comment]
Actually I can't open an issue. Please, can someone help me find out how they fixed it? Thanks

EDIT: @arthur-esob @mclien Was it a quick fix? I'm digging into this and am completely lost.

**Comment by Graystripe17** [comment]
It's ok y'all. My problem was the exact same as @arthur-esob, I have the RPi2B
I copied neopixel.py into examples `sudo cp neopixel.py /examples` and tested it out there where it couldn't find module _rpi_ws281x. For some reason it started working when I ran `sudo python setup.py install`. I thought I had previously installed it, so run it again just to be sure.

Don't forget to run 
`sudo apt-get install python-dev swig`
`python ./setup.py build`
as per the README.

May everyone have beautiful lights without problem!
